### PR TITLE
mirgrate loadpaths to harmony.

### LINKED ISSUE #2: migrate LoadPaths to harmony
I am trying to use harmony to patch LoadPaths to apply TMPE rules set in asset editor.
It seems to be easy to migrate this method to harmony.

### PR DESCRIPTION
**Pull request by kianzarrin**
fixes #2 
- installed cities harmony. migrated `LoadPaths()` to harmony.
- harmony installation is not as fast as detours. so I think it is best to install/uninstall it in loading extension. smart intersection builder is not active, harmony prefix/postfix/... return silently.
- bumped version 
- version is centrally controlled by assembly version. this helps other mods to check if they are compatible with smart intersection builder.
I copy-pasted my transpiler utils from my other project here. There are some methods over there that I have not used. should I remove them or keep them for future use?

I replaced Load paths redirection with a prefix and a transpiler.

the following methods were copy-pasted from `BuildingsDecorationsDetour`:
- `ReleaseCollidingSegments`
- `AfterIntersectionBuilt`
- `ReleaseQuestionableSegments`

The following methods were deleted because they were exact copies of stock code (I assume you copy-pasted them because you couldn't be bothered to use reflection to involve private methods.). they are no longer necessary not that we are using harmony.
- `RequireFixedHeight`
- `TrafficLightsToFlags`
- `FindConnectNode`

### CODE REVIEW

**Comment by kianzarrin** [comment]
I will test this tommorow.

**Comment by kianzarrin** [comment]
@Strdate my code destroys the conflicting segments but does not build new segments. do you know what am I missing?
image
image

**Comment by Strdate** [comment]
Hi, I will check this out later this week. Thanks for your work.
There were some kind of reports from the steam workshop page that the roads don't connect, but I don't know it this is the same problem. I wasn't able to reproduce this anyway.

**Comment by kianzarrin** [comment]
thanks.
I checked and all the prefixes and postfixes are being called. but I still get this.
image
It is going to be hard for me to debug a code I do not understand. would you like to take over from here?

**Comment by Strdate** [comment]
Yeah, it's ok :)

File: SmartIntersections/Patch/LoadPathsPatch.cs [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+                }
+            }
+
+            borderNodes.RemoveWhere(n => !NetUtil.ExistsNode(n.Node));
+
+            ToolControllerDetour.CollidingSegmentsCache2 = null;
+
+            //Debug.Log("Border nodes (1): " + borderNodes.Count);
+
+            borderNodes = borderNodes;
```
Comment (Strdate) [inline SmartIntersections/Patch/LoadPathsPatch.cs:74]: There is the problem
(I've already fixed it)

**Comment by Strdate** [comment]
Merged

**Comment by Strdate** [comment]
Can you please test the workshop version?

**Comment by kianzarrin** [comment]
Excelent!

File: SmartIntersections/Patch/LoadPathsPatch.cs (hunk 1 above)
Comment (kianzarrin) [reply SmartIntersections/Patch/LoadPathsPatch.cs:74]: Oh god! wow! so it was the scope! I should learn my lesson so as to avoid such a mistake in future.

**Comment by kianzarrin** [comment]
I can confirm that this is working with my node controller which does patch LoadPaths to map node IDs. 
Screenshot (1138)
Screenshot (1139)
Note that the corner offset(smooth intersection) is applied which means my node controller data was copy-pasted from the asset.

**Pull request merged**